include/aspect/introspection.h

```diff
@@ -465,6 +465,21 @@ namespace aspect
        * @}
        */
 
+      /**
+       * Return a vector with all used indices of base elements of the deal.II FiniteElement
+       * that are used for compositional fields. If several fields are the same type, they
+       * share base elements.
+      */
+      std::vector<unsigned int>
+      get_compositional_field_base_element_indices() const;
+
+      /**
+       * Return a vector with all compositional field indices that belong to a given
+       * base element index.
+      */
+      std::vector<unsigned int>
+      get_compositional_field_indices_with_base_element(const unsigned int base_element_index) const;
```

Comment:
Is this indexed starting at the zeroth compositional element? Or at the velocity? 

Or perhaps you are saying that the argument must be an element of the array that `get_compositional_field_base_element_indices()` returns? If so, please add that. (And assert?)